pkg/k8s/service.go

```diff
@@ -98,7 +110,21 @@ func ParseService(svc *slim_corev1.Service, nodeAddressing datapath.NodeAddressi
 		return ServiceID{}, nil
 	}
 
-	clusterIP := net.ParseIP(svc.Spec.ClusterIP)
+	var clusterIPs []net.IP
+	if len(svc.Spec.ClusterIPs) == 0 {
+		if clsIP := net.ParseIP(svc.Spec.ClusterIP); clsIP != nil {
+			clusterIPs = []net.IP{clsIP}
+		}
+	} else {
+		// Here we assume that the value of .spec.ClusterIPs[0] is same as that of the .spec.clusterIP
+		// or else Kubernetes will reject the service with validation error.
+		for _, ip := range svc.Spec.ClusterIPs {
```

Comment:
I think we can simplify this block:

```suggestion
	clusterIPs := make([]net.IP, 0, len(svc.Spec.ClusterIPs))
	if clsIP := net.ParseIP(svc.Spec.ClusterIP); clsIP != nil {
		clusterIPs = append(clusterIPs, clsIP)
	}
	// Here we assume that the value of .spec.ClusterIPs[0] is same as that of the .spec.clusterIP
	// or else Kubernetes will reject the service with validation error.
	for _, ip := range svc.Spec.ClusterIPs {
```

Reply:
That will cause `clusterIPs` to have duplicated IP addresses. (`svc.Spec.ClusterIP` == `svc.Spec.ClusterIPs[0]`)

Reply:
Ah good catch